#!/usr/bin/env python

PACKAGE="realsense_camera"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

#             Name                               Type      Level       Description                     Default    Min     Max
gen.add("enable_depth",                          bool_t,   0,          "Enable Depth",                 True)
gen.add("depth_control_preset",                  int_t,    0,          "Depth Control Preset",         5,         0,      5)
gen.add("color_backlight_compensation",          int_t,    0,          "Backlight Compensation",       1,         0,      4)
gen.add("color_brightness",                      int_t,    0,          "Brightness",                   56,        0,      255)
gen.add("color_contrast",                        int_t,    0,          "Contrast",                     32,        16,     64)
gen.add("color_gain",                            int_t,    0,          "Gain",                         32,        0,      256)
gen.add("color_gamma",                           int_t,    0,          "Gamma",                        220,       100,    280)
gen.add("color_hue",                             int_t,    0,          "Hue",                          0,         -2200,  2200)
gen.add("color_saturation",                      int_t,    0,          "Saturation",                   128,       0,      255)
gen.add("color_sharpness",                       int_t,    0,          "Sharpness",                    0,         0,      7)

# Must be set only if color_enable_auto_white_balance is disabled
gen.add("color_white_balance",                   int_t,    0,          "White Balance",                6500,      2000,   8000)

gen.add("color_enable_auto_white_balance",       int_t,    0,          "Enable Auto White Balance",    1,         0,      1)
gen.add("r200_lr_auto_exposure_enabled",         int_t,    0,          "LR Auto Exposure Enabled",     0,         0,      1)
gen.add("r200_lr_gain",                          int_t,    0,          "LR Gain",                      400,       100,    6399)

# Must be set only if r200_lr_auto_exposure_enabled is disabled
gen.add("r200_lr_exposure",                      int_t,    0,          "LR Exposure",                  164,       1,      164)

gen.add("r200_emitter_enabled",                  int_t,    0,          "Emitter Enabled",              1,         0,      1)

# The following edge parameters must be set only if r200_lr_auto_exposure_enabled is enabled
gen.add("r200_auto_exposure_top_edge",           int_t,    0,          "Auto Exposure Top Edge",       0,         0,      479)
gen.add("r200_auto_exposure_bottom_edge",        int_t,    0,          "Auto Exposure Bottom Edge",    479,       0,      479)
gen.add("r200_auto_exposure_left_edge",          int_t,    0,          "Auto Exposure Left Edge",      0,         0,      639)
gen.add("r200_auto_exposure_right_edge",         int_t,    0,          "Auto Exposure Right Edge",     639,       0,      639)

exit(gen.generate(PACKAGE, "realsense_camera", "r200_params"))
